tests/test_tree.py

```diff
@@ -31,6 +31,22 @@ def test_tree_at_replace(getkey):
     assert jnp.all(pytree2[-1].weight == true_pytree2[-1].weight)
     assert jnp.all(pytree2[-1].bias == true_pytree2[-1].bias)
 
+    pytree = [(), (), {"a": jnp.array([1.0, 2.0])}, eqx.nn.Linear(1, 2, key=key1)]
+    true_pytree1 = [1, (), {"a": "hi"}, eqx.nn.Linear(1, 2, key=key1)]
+    where1 = lambda tree: tree[0]
+    weight2 = true_pytree2[3].weight
+    bias2 = true_pytree2[3].bias
+    pytree1 = eqx.tree_at(where1, pytree, replace=1)
+
+    assert pytree1[:-2] == true_pytree1[:-2]
+    assert pytree2[:-2] == true_pytree2[:-2]
+    assert jnp.all(pytree1[-2]["a"] == true_pytree1[-2]["a"])
+    assert jnp.all(pytree2[-2]["a"] == true_pytree2[-2]["a"])
+    assert jnp.all(pytree1[-1].weight == true_pytree1[-1].weight)
+    assert jnp.all(pytree1[-1].bias == true_pytree1[-1].bias)
+    assert jnp.all(pytree2[-1].weight == true_pytree2[-1].weight)
+    assert jnp.all(pytree2[-1].bias == true_pytree2[-1].bias)
+
```

Comment:
I think let's pull this out as a separate test, it's edge case enough to want to highlight things. Something like this is probably what we want I think:
```python
a = ()
b= (1,)
x1 = [a]
x2 = [a, a]
x3 = [(), ()]
x4 = [b]
x5 = [b, b]
x6 = [(1,), (1,)]

for x in (x1, x2, x3, x4, x5, x6):
    new_x = eqx.tree_at(lambda xi: xi[0], x, "hello")
    assert new_x[0] == "hello"
    if len(new_x) != 1:
        assert new_x[1] != "hello"
```

WDYT?

Reply:
Sure, makes sense, pulled it out